reference/src/glossary.md

```diff
@@ -73,6 +73,12 @@ niches. For example, the "all bits uninitialized" is an invalid bit-pattern for
 `&mut T`, but this bit-pattern cannot be used by layout optimizations, and is not a
 niche.
 
+#### Zero-sized type / ZST
+
+Types with zero size are called zero-sized types, which is abbreviated as "ZST".
+This document also uses the "1-ZST" abbreviation, which stands for "one-aligned
+zero-sized type", to refer to zero-sized types with an alignment requirement of
+1.
```

Comment:
Might be worth mentioning (because this is not terribly widely known) that `[u32; 0]` is an example of a ZST that is *not* a 1-ZST.